### PR TITLE
Adding alert context

### PR DESCRIPTION
**Pull request by fangyu3**
Adding alert in login page

Adding alert to onboarding

Adding alert to event components

### CODE REVIEW

**Comment by fangyu3** [comment]
@sina-jamshidi can we prioritize this PR as it fixes the authentication bug

File: client/src/components/EventTypeDialog.js [hunk 1]
```diff
@@ -102,17 +113,17 @@ export default function EventTypeDialog() {
             .then(res => {
                 status = res.status;
                 if (status < 500) return res.json();
-                else throw Error("Server error");
+                else setAlert(true, "Server error", "error")
             })
```
Comment (JoshL579) [inline client/src/components/EventTypeDialog.js]: I think we can keep these `throw Error` here and deal with all of them at the end in `.catch`

Comment (fangyu3) [reply client/src/components/EventTypeDialog.js]: Good point!

File: client/src/components/Onboarding.js [hunk 2]
```diff
@@ -76,34 +76,54 @@ function Onboarding({children,title,activeStep,path,url}) {
...
+                            return setAlert(true, "Url already exists", "error")
                         };
-                        return alert("Url exist")
-                    };
-                    throw Error("Failed to check");                
-                })
-                .catch(err => {
-                    alert(err.message);
-                });
+                        throw setAlert(true, "Unexpected error", "error")
```
Comment (johnsciutto) [inline client/src/components/Onboarding.js]: Watch out here, you're throwing `setAlert`. I don't know if this is what you intended.

**Review by johnsciutto — COMMENTED** [review]
You could probably dispense with the `setAlert` functions you're creating in all the different components where you want to update the alert context, and use the `setAlertStatus` directly. 

If you really want to have that function to act as a mediator to update the `alertStatus`, then add it to the `AlertContextProvider` and export it as part of the `value`.

That way, you save yourself from having to re-write it in every `AlertContext` consumer.

**Review by sina-jamshidi — APPROVED** [review]
I agree with John's comment. Afterwards this is good to go

**Pull request merged**